src/include/cephfs/metrics/Types.h

```diff
@@ -367,6 +367,10 @@ struct UnknownPayload {
   }
 
   void decode(bufferlist::const_iterator &iter) {
+    using ceph::decode;
```

Comment:
This looks correct.

The original decision was to abort if an unknown metric is received. Then we wanted to handle clients that won't send any metrics for backward compat.

Reply:
Okay.
I just used 254 instead of 255 below to dismiss the following warning:

```
In file included from /data/ceph/src/include/uuid.h:9,
                 from /data/ceph/src/include/types.h:21,
                 from /data/ceph/src/include/utime.h:28,
                 from /data/ceph/src/common/LogEntry.h:18,
                 from /data/ceph/src/common/LogClient.h:19,
                 from /data/ceph/src/mds/MDSRank.h:23,
                 from /data/ceph/src/mds/SessionMap.cc:15:
/data/ceph/src/include/cephfs/metrics/Types.h: In member function ‘void UnknownPayload::decode(ceph::buffer::v15_2_0::list::const_iterator&)’:
/data/ceph/src/include/encoding.h:1389:9: warning: comparison is always false due to limited range of data type [-Wtype-limits]
   if (v < struct_compat)      \
/data/ceph/src/include/cephfs/metrics/Types.h:318:5: note: in expansion of macro ‘DECODE_START’
     DECODE_START(255, iter);
     ^~~~~~~~~~~~
```

Reply:
I suppose you _always_ want to throw `::ceph::buffer::malformed_input` ? Why not just do that directly?

Reply:
> I suppose you _always_ want to throw `::ceph::buffer::malformed_input` ? Why not just do that directly?

Do you mean when the MDS daemon receives an unkown type of metric ? Just throw `::ceph::buffer::malformed_input` in the `UnknownPayload::decode()` ?

This change will just try to drop the unknown type of metric and ignore them and continue if the kclient is using a higher version, ususally the buffer contains correct binary data, so ignoring the unkown metircs should be no harm. 

Only in rare case that the binary is really corrupted and in this case it'll probably throw `::ceph::buffer::malformed_input` just like this issue we hit, if not it may corrupt  known type of metrics, but since it is rare it should be acceptable for the cephfs-top show (?).

Or should we just throw `::ceph::buffer::malformed_input` in the `UnknownPayload::decode()` ? Will it be better ?

Reply:
Ah, I see. No, what you have makes sense.